### ISSUE TITLE
Permission denied with smCOG

### ISSUE THREAD

**Issue by cjfields** [antismash/antismash#157]
**Describe the bug**

I am seeing a 'Permission denied' error with using the `--smcog-trees` parameter; tested using a manual installation of antiSMASH and a `venv` installation.   The application finishes successfully without this parameter.

```
[cjfields@compute-8-0 2019-05-16-test-as]$ antismash --logfile as.log --debug --cb-general --cb-knownclusters --cb-subclusters --asf --pfam2go --smcog-trees -c $SLURM_NPROCS --genefinding-gff GCF_002157265.1_ASM215726v1_genomic.gff GCF_002157265.1_ASM215726v1_genomic.fna
...
DEBUG    16/05 10:40:44   annotating CDS features with resist info: 1 CDSes
DEBUG    16/05 10:40:44   annotating CDS features with smcogs info: 174 CDSes
DEBUG    16/05 10:40:44   Checking if antismash.modules.smcog_trees should be run
INFO     16/05 10:40:44   Running antismash.modules.smcog_trees
INFO     16/05 10:40:44   Calculating and drawing phylogenetic trees of cluster genes with smCOG members
multiprocessing.pool.RemoteTraceback:
"""
Traceback (most recent call last):
  File "/home/apps/software/Python/3.6.1-IGB-gcc-4.9.4/lib/python3.6/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/apps/software/Python/3.6.1-IGB-gcc-4.9.4/lib/python3.6/multiprocessing/pool.py", line 47, in starmapstar
    return list(itertools.starmap(args[0], args[1]))
  File "/home/groups/hpcbio/apps/antismash/antismash-github/install/lib/python3.6/site-packages/antismash/modules/smcog_trees/trees.py", line 68, in smcog_tree_analysis
    draw_tree(input_number, output_dir, gene_id)
  File "/home/groups/hpcbio/apps/antismash/antismash-github/install/lib/python3.6/site-packages/antismash/modules/smcog_trees/trees.py", line 138, in draw_tree
    run_result = subprocessing.execute(command)
  File "/home/groups/hpcbio/apps/antismash/antismash-github/install/lib/python3.6/site-packages/antismash/common/subprocessing/base.py", line 86, in execute
    proc = Popen(commands, stdin=stdin_redir, stdout=stdout, stderr=stderr)
  File "/home/apps/software/Python/3.6.1-IGB-gcc-4.9.4/lib/python3.6/subprocess.py", line 707, in __init__
    restore_signals, start_new_session)
  File "/home/apps/software/Python/3.6.1-IGB-gcc-4.9.4/lib/python3.6/subprocess.py", line 1326, in _execute_child
    raise child_exception_type(errno_num, err_msg)
PermissionError: [Errno 13] Permission denied
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/groups/hpcbio/apps/antismash/antismash-github/install/bin/antismash", line 11, in <module>
    sys.exit(entrypoint())
  File "/home/groups/hpcbio/apps/antismash/antismash-github/install/lib/python3.6/site-packages/antismash/__main__.py", line 124, in entrypoint
    sys.exit(main(sys.argv[1:]))
  File "/home/groups/hpcbio/apps/antismash/antismash-github/install/lib/python3.6/site-packages/antismash/__main__.py", line 113, in main
    antismash.run_antismash(sequence, options)
  File "/home/groups/hpcbio/apps/antismash/antismash-github/install/lib/python3.6/site-packages/antismash/main.py", line 574, in run_antismash
    result = _run_antismash(sequence_file, options)
  File "/home/groups/hpcbio/apps/antismash/antismash-github/install/lib/python3.6/site-packages/antismash/main.py", line 638, in _run_antismash
    analysis_timings = analyse_record(record, options, analysis_modules, module_results)
  File "/home/groups/hpcbio/apps/antismash/antismash-github/install/lib/python3.6/site-packages/antismash/main.py", line 264, in analyse_record
    run_module(record, module, options, previous_result, timings)
  File "/home/groups/hpcbio/apps/antismash/antismash-github/install/lib/python3.6/site-packages/antismash/main.py", line 236, in run_module
    results = module.run_on_record(record, results, options)
  File "/home/groups/hpcbio/apps/antismash/antismash-github/install/lib/python3.6/site-packages/antismash/modules/smcog_trees/__init__.py", line 128, in run_on_record
    trees = generate_trees(smcogs_dir, record.get_cds_features_within_regions(), nrpspks_genes)
  File "/home/groups/hpcbio/apps/antismash/antismash-github/install/lib/python3.6/site-packages/antismash/modules/smcog_trees/trees.py", line 48, in generate_trees
    subprocessing.parallel_function(smcog_tree_analysis, args)
  File "/home/groups/hpcbio/apps/antismash/antismash-github/install/lib/python3.6/site-packages/antismash/common/subprocessing/base.py", line 132, in parallel_function
    results = jobs.get(timeout=timeout)
  File "/home/apps/software/Python/3.6.1-IGB-gcc-4.9.4/lib/python3.6/multiprocessing/pool.py", line 608, in get
    raise self._value
PermissionError: [Errno 13] Permission denied
```
**Expected behavior**

Well, that `--smcog-trees` would work, but I know this one is a difficult issue to wrangle, particularly with a complex tool with a lot of moving parts ;)

**To Reproduce**

Example data (`GCF_002157265.1_ASM215726v1_genomic.fna` and `GCF_002157265.1_ASM215726v1_genomic.gff`) from ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/002/157/265/GCF_002157265.1_ASM215726v1

Using Github checkout cbd4ad31 (May 14).

**Screenshots**

None

**System (please complete the following information):**
 - OS:

[cjfields@compute-8-0 2019-05-16-test-as]$ cat /etc/redhat-release
CentOS Linux release 7.6.1810 (Core)

 - Browser [e.g. chrome, safari] (if relevant)

None (command line)

**Additional context**

The run passes all checks for installation; I am attaching a log file for this

My suspicion is that an application is being called that isn't installed, but I can't hone in on which one it is.  `muscle` or `Fasttree` are in the path.
[as.log](https://github.com/antismash/antismash/files/3188049/as.log)

**Comment by SJShaw** [comment]
From your logfile:
```INFO     16/05 10:39:11   muscle using executable: /home/apps/software/MUSCLE/3.8.31-IGB-gcc-4.9.4/bin/muscle (v3.8.31)
INFO     16/05 10:39:11   fasttree using executable:```
So it looks like it's `fasttree`, not finding it due to the case of the `F`.

This should be caught in the prerequisite checks that run, I'll look into that as it shouldn't get this far without being able to find a binary dependency.

In the meantime, if you try running with the extra option `--executable-paths fasttree=Fasttree`, you'll hopefully see it function as expected.

**Comment by cjfields** [comment]
@SJShaw thanks, I missed that (I assumed the check would catch it).  Yes, this seemed to address the problem, it has completed that stage w/o issues and I can see `FastTree` being called in the background now.

**Comment by kblin** [comment]
Thanks for the report, Chris. I've just merged the pull request and will close the issue once we have a release containing it. If you installed from the git repo, you should already be able to update to get the fix.

**Comment by kblin** [comment]
Fixed in our 5.0.0 release

**Comment by cjfields** [comment]
Thanks @kblin !
